agent/dockerclient/dockerapi/docker_client.go

```diff
@@ -392,6 +388,10 @@ func (dg *dockerGoClient) pullImage(ctx context.Context, image string, authData
 				pullFinished <- err
 				return
 			}
+			if data.Error != "" {
```

Comment:
can we add at least unit test coverage for this new code path?

Reply:
sure, will do